### DISCUSSION TITLE
Querying data with the Supabase Client and foreign tables of different schemas

### DISCUSSION THREAD

**Discussion by bombillazo** [supabase/supabase#14748] [Questions]
Hey, a common pattern is to store the user id and basic data in the auth schema, users table. However, the main logic and project data lives in public. If we have foreign relations from our public schema tables to auth.users, how can we query those foreign tables with the supabase client?

```ts
// does not wotk
const { data, error } = await supabase
  .from('profile')
  .select('name, auth.users(email)')
  .eq('auth.users.id', user_id)
```

**Comment by bombillazo** [comment]
Duplicate of https://github.com/orgs/supabase/discussions/12997

**Comment by steve-chavez** [comment]
`auth.users` contains private information. You have to create a VIEW of `auth.users` under the public schema that only exposes the columns you want external users to see. Then you can use it normally in the supabase client.
